setuptools/dist.py

```diff
@@ -702,30 +703,45 @@ def finalize_options(self):
         to influence the order of execution. Smaller numbers
         go first and the default is 0.
         """
+
+        default_order_value = 0
         hook_key = 'setuptools.finalize_distribution_options'
 
         def by_order(hook):
-            return getattr(hook, 'order', 0)
+            return getattr(hook, 'order', default_order_value)
+
         eps = pkg_resources.iter_entry_points(hook_key)
+
         for ep in sorted(eps, key=by_order):
-            ep.load()(self)
+            try:
+                f = ep.load()
+            except Exception as ex:
```

Comment:
This branch probably needs a test. And it probably deserves to be its own function with a docstring describing its purpose.

Reply:
I am not sure how to properly test this. If we move that into a separate function and test only it, it will turn out we are actually testing `EntryPoint.load`. To test it properly I guess we need to mock `iter_entry_points` into a one returning our not really existing entry point.